File: connectx-backend/src/main/java/com/connectx/group/dto/CreateGroupInvitationResponseDto.java
package com.connectx.group.dto;

/**
 * outcome is one of "DIRECT_ADDED" (target was added to the group immediately -- invitation is
 * null) or "INVITATION_SENT" (a PENDING GroupInvitation was created -- invitation is populated).
 * A DENIED decision never reaches this DTO; it surfaces as a thrown ApiException instead.
 */
public class CreateGroupInvitationResponseDto {

    private String outcome;
    private GroupInvitationDto invitation;

    public CreateGroupInvitationResponseDto() {}

    public CreateGroupInvitationResponseDto(String outcome, GroupInvitationDto invitation) {
        this.outcome = outcome;
        this.invitation = invitation;
    }

    public String getOutcome() {
        return outcome;
    }

    public void setOutcome(String outcome) {
        this.outcome = outcome;
    }

    public GroupInvitationDto getInvitation() {
        return invitation;
    }

    public void setInvitation(GroupInvitationDto invitation) {
        this.invitation = invitation;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/group/model/CreateGroupInvitationResponseDto.kt
package com.connectx.app.data.remote.group.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.group.dto.CreateGroupInvitationResponseDto exactly.
 * `outcome` is one of "DIRECT_ADDED" (target added immediately -- `invitation` is
 * null) or "INVITATION_SENT" (`invitation` is populated). A DENIED decision never
 * reaches this DTO on the backend -- it surfaces as an error response instead, so
 * no third outcome value is modeled here.
 */
@Serializable
data class CreateGroupInvitationResponseDto(
    val outcome: String? = null,
    val invitation: GroupInvitationDto? = null
)
